src/flupy/fluent.py

```diff
@@ -475,6 +475,12 @@ def map_attr(self, attr: str) -> "Fluent[Any]":
         """
         return self.map(lambda x: getattr(x, attr))
 
+    @overload
+    def filter(self, func: Callable[[T], bool], *args: Any, **kwargs: Any) -> "Fluent[T]": ...
+
+    @overload
+    def filter(self, func: Callable[..., bool], *args: Any, **kwargs: Any) -> "Fluent[T]": ...
```

Comment:
is this one needed since its the same as the definition containing the body?

Reply:
I don't know. I tried without it, and mypy (version 1.10.0) gave the error `Single overload definition, multiple required`.

The [overload examples in mypy's docs](https://mypy.readthedocs.io/en/stable/more_types.html#function-overloading) don't have any exact analogs. I'm happy to search elsewhere, just let me know. :slightly_smiling_face: 

----

Incidentally, I think that the **first** overload should omit `args` and  `kwargs`, since the function is specified as having only one argument. That is,

```python3
@overload
def filter(self, func: Callable[[T], bool]         ) -> "Fluent[T]": ...
#                                           ^^^removed from here
```

Thoughts?

Reply:
After doing a bit more research, it's possible that [Callback Protocols](https://mypy.readthedocs.io/en/stable/protocols.html#callback-protocols) would help. More to follow either tomorrow or next week.

Reply:
Ok, so I think I figured it out, no overloads needed! I didn't test this thoroughly, but I'm putting it here to see if you have thoughts. Also, please feel free to rename anything.

```python3
T = TypeVar("T")
Tret = TypeVar("Tret")

class FluentFunc(Protocol[T, Tret]):
    def __call__(self, x: T, *args, **kwargs) -> Tret: ...


class Fluent(Generic[T]):
    def filter(self, f: FluentFunc[T, bool], *args, **kwargs):
        # implementation here
```